include/common/serialise_matrix.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+// BoB robotics includes
+#include "macros.h"
+
+// Third-party includes
+#include "third_party/path.h"
+
+// Eigen
+#include <Eigen/Core>
+
+// Standard C++ includes
+#include <fstream>
+
+namespace BoBRobotics {
+//! Read a matrix of type T from a binary file
+template<class T>
+auto
+readMatrix(const filesystem::path &filepath)
+{
+    // Open file
+    std::ifstream ifs;
+    ifs.exceptions(std::ios::badbit | std::ios::eofbit);
+    ifs.open(filepath.str(), std::ios::in | std::ios::binary);
+
+    // The matrix size is encoded as 2 x int32_t
+    int32_t size[2];
+    ifs.read(reinterpret_cast<char *>(&size), sizeof(size));
+
+    // Create data array and fill it
+    using namespace Eigen;
+    Matrix<T, Dynamic, Dynamic> data(size[0], size[1]);
+    ifs.read(reinterpret_cast<char *>(data.data()), sizeof(T) * data.size());
+
+    return data;
+}
+
+template<class T>
+void
+writeMatrix(const filesystem::path &filepath, const T &matrix)
```

Comment:
Might be a little safer to template this on:
```c++
template<class T, int R, int C>
void 
writeMatrix(const filesystem::path &filepath, const Matrix<T, R, C> &matrix)
```
(or whatever template arguments Matrix has)

Reply:
Good idea. It would also be more descriptive that way.

Reply:
Actually it seems like this won't really work because Eigen's templated types can end up really complicated, e.g. I'm wrapping around another data structure somewhere and end up with a ``Eigen::Map<Eigen::Matrix<float, -1, 1>, 0, Eigen::Stride<0, 0> >`` which won't work with your suggestion :cry:. (You'd think it'd end up being implicitly casted, but apparently not!)